src/main/java/io/vertx/core/http/impl/ClientConnection.java

```diff
@@ -391,8 +393,8 @@ protected synchronized void handleClosed() {
       for (HttpClientRequestImpl req: requests) {
         metrics.requestReset(req.metric());
       }
-      if (requestForResponse != null) {
-        metrics.requestReset(requestForResponse.metric());
+      if (currentResponse != null) {
```

Comment:
bug fix yes - was calling endRequest in metrics twice when connection was closed after response sent.